presto-main/src/main/java/io/prestosql/spiller/SpillCipher.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spiller;
+
+import java.util.Arrays;
+
+public interface SpillCipher
+        extends AutoCloseable
+{
+    /**
+     * Calculates the maximum required buffer size to encrypt input data with the given length
+     */
+    int encryptedMaxLength(int inputLength);
+
+    /**
+     * Encrypts the contents of the input buffer into the destination buffer returning the number
+     * of bytes written into the destination buffer
+     * @return The length of the encrypted content in the destination buffer
+     */
+    int encrypt(byte[] data, int inputOffset, int length, byte[] destination, int destinationOffset);
+
+    default byte[] encrypt(byte[] data)
```

Comment:
I believe this is unused, so I'd prefer to drop it because it can result in an extra copy.... also we generally don't like to keep around unused code.

Reply:
I see this is used in the test.  Let's move it there.

Reply:
Sure, will move to a static helper method in test.